### DISCUSSION TITLE
Only allow admin user to add new websites

### DISCUSSION THREAD

**Discussion by tpimh** [plausible/analytics#1906] [Anything Else]
Currently, with publicly exposed self-hosting Plausible instances, where registration and login are not disabled, it's possible to register new user accounts, add any websites, and use the dashboard without any limits.

It would be nice to have a setting like `USERS_CAN_ADD_WEBSITES`. When this setting is set to `FALSE` (I suggest this should be the default), anyone can still register, but only the `ADMIN_USER` can add new websites (other users can be invited to become Admins or Viewers of these sites, but can never become Owners). In my opinion, this behavior will prevent the unwanted external users from exploiting the self-hosted installations.

**Comment by arrigolupori** [comment]
This is important for me as well!

Even better, remove the option for other users to sign up altogether.
Since it's a self-hosted install, I don't expect external users to sign up, it's a security risk.

Is there any way that we can implement this in self-hosted?

Thank you!

**Reply by madanyang** [reply]
Add the following to your environment configuration
```
 DISABLE_REGISTRATION=true
```

You can read more about the available options in [Configuration options](https://plausible.io/docs/self-hosting-configuration)

**Reply by arrigolupori** [reply]
Thank you, I did not see this at all in the docs!

Really helpful and easy to configure :)

**Reply by jesperordrup** [reply]
This prevents people from creating accounts but when they have an account they can still create a website.

**Comment by stefanv** [comment]
Does this actually solve the problem? Now, when I try and add users to a site, they cannot register, even when I invite them.

**Comment by stefanv** [comment]
Solution: https://github.com/plausible/analytics/issues/1706#issuecomment-1376755032

**Comment by jesperordrup** [comment]
This is still a valid request #1706 doesnt solve it.

First, theres a difrerence between allowing registrations (false/true/invite_only) and access to creating websites.

It seems theres a need for 
ADD_WEBSITES=    all/users/admins/...

Im not sure if its actually solved already ?

**Reply by ruslandoga** [reply]
👋 @jesperordrup 

> Im not sure if its actually solved already ?

I don't think there is an option to turn off the ability to create websites yet.

**Reply by jesperordrup** [reply]
ok, is this the best way to suggest or do I need to describe elsewhere?

**Reply by ruslandoga** [reply]
I think the feature request is clear :) I'll ask the core team on the next call if there are any plans for this to be implemented and report back!

**Reply by jesperordrup** [reply]
@ruslandoga any updates on what the team said?

**Reply by ruslandoga** [reply]
👋 @jesperordrup 

Sorry, I don't work on CE anymore.

IIRC we added it to the backlog at the time but never got around to implementing it.

**Reply by jesperordrup** [reply]
Well thats what happens. I imagine that since this is something only relevant for the community - it will never happen. Unless in a fork.

So the fork hunt begins 💪😁

**Comment by RasberryKai** [comment]
Hey is there a solution for this already, because I need it for my project?

**Reply by ruslandoga** [reply]
👋 @RasberryKai 

No, right now all registered users can add websites.

**Reply by jesperordrup** [reply]
@RasberryKai We forked and made a temporary fix. Just hidding the button. Clearly not enough but ok because we know the users.

![image](https://github.com/user-attachments/assets/3bf8319e-b85a-422d-a3e8-084a8b31ece2)

**Comment by jesperordrup** [comment]
Preventing users from creating websites based on role or env or anything? 

I think its a very commen need for the community edition, should we revive the issue?
